cvat/apps/engine/views.py

```diff
@@ -864,6 +870,8 @@ def get_queryset(self):
         if self.action == 'list':
             perm = TaskPermission.create_scope_list(self.request)
             queryset = perm.filter(queryset)
+        elif self.action == 'preview':
+            queryset = Task.objects.filter(**self.kwargs).select_related('data')
```

Comment:
I believe you should also rely here on a queryset obtained from `super().get_queryset()`. + will it fail if `self.kwargs` contains fields that do not exist in Task model?

Reply:
> I believe you should also rely here on a queryset obtained from super().get_queryset(). 

The idea itself is not to rely on default queryset, as it performs several additional requests to the database with joins. All these data are not necessary when we only need preview. 

`.prefetch_related(
        'segment_set__job_set',
        'segment_set__job_set__assignee',
    ).with_job_summary().all()
`

Reply:
But filtering was extra here. 
Django will apply filter by `pk` in `self.get_object`.